src/core/Akka.Cluster/SBR/DowningStrategy.cs

```diff
@@ -256,6 +260,51 @@ private ImmutableHashSet<UniqueAddress> AdditionalNodesToDownWhenIndirectlyConne
             }
         }
 
+        #region Unreachable island detection
+        /// <summary>
+        /// Detect floating unreachable islands in the reachability graph.
+        /// 
+        /// Note, this is a potentially expensive operation to perform, this is designed to be called by
+        /// `AdditionalNodesToDownWhenIndirectlyConnected()` which is very rarely called, do not call this anywhere else.
+        /// </summary>
+        /// <param name="reachability">Local copy of the reachability graph</param>
+        /// <param name="unreachable">Local copy of the unreachable addresses</param>
+        /// <returns>Hash set of detected disconnected addresses</returns>
+        private ImmutableHashSet<UniqueAddress> PruneUnreachableNodes(
+            Reachability reachability,
+            ImmutableHashSet<UniqueAddress> unreachable)
+        {
+            // Get all reachable nodes
+            var allReachable = AllMembers.Select(m => m.UniqueAddress)
+                .Where(a => !unreachable.Contains(a))
+                .ToImmutableHashSet();
```

Comment:
From the all cluster members set, exclude all members that are inside the unreachable set. This gives us all of the cluster members that is reachable.